code/datums/gamemode/factions/spider_clan.dm

```diff
@@ -0,0 +1,160 @@
+/datum/faction/spider_clan
+	name = "Spider Clan"
+	desc = "For honor, for revengeance, or just to train by ruining someone's day."
+	ID = SPIDERCLAN
+	required_pref = NINJA
+	initial_role = NINJA
+	late_role = NINJA
+	roletype = /datum/role/ninja
+	initroletype = /datum/role/ninja
+	logo_state = "ninja-logo"
+	hud_icons = list("ninja-logo")
+
+/datum/faction/spider_clan/New()
+	..()
+	load_dungeon(/datum/map_element/dungeon/ninja_dojo)
+
+
+/datum/faction/spider_clan/forgeObjectives()
+	if(cyborg_list.len)
+		AppendObjective(/datum/objective/target/killsilicons)
+	else
+		if(prob(70))
+			AppendObjective(/datum/objective/target/delayed/assassinate)
+		else
+			AppendObjective(/datum/objective/target/skulls)
+
+	if(ai_list.len)
+		AppendObjective(/datum/objective/killorstealAI)
+	else
+		AppendObjective(/datum/objective/target/steal)
+
+	var/living = 0
+	for(var/mob/living/M in player_list)
+		if(!M.client)
+			continue
+		if(!iscarbon(M) && !issilicon(M))
+			continue
+		var/turf/T = get_turf(M)
+		if(T.z != STATION_Z)
+			continue
+		if(M.stat != DEAD)
+			living++
+	if(living<=16 && prob(25))
+		AppendObjective(/datum/objective/silence)
+	else
+		AppendObjective(/datum/objective/survive)
+	if(prob(15))
+		AppendObjective(/datum/objective/stealsake)
```

Comment:
didn't you put sake in the dojo

Reply:
yes but they have to survive an entire round with it AND sake heals ninjas